.github/workflows/browser-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: Browser Tests
+
+on:
+    pull_request:
+        types: [opened, synchronize, reopened]
+
+jobs:
+    compare-firefox:
+        name: Compare performance to latest release on Firefox
+
+        # The job will only run if the pull request is from the same repository.
+        # Benchmarks can't run on PRs from forked repos due to comment posting restrictions without a GitHub token.
+        if: github.event.pull_request == null || github.event.pull_request.head.repo.full_name == github.repository
+        runs-on: ubuntu-22.04
+        steps:
+            - name: Checkout main
+              uses: actions/checkout@v4
+              with:
+                  ref: main
+
+            - name: Checkout PR branch
+              uses: actions/checkout@v4
+
+            - name: Setup Job and Install Dependencies
+              uses: ./.github/actions/setup-job
+
+            - name: Print Firefox version
+              run: firefox --version
+
+            - name: Tachometer the changed packages
+              run: yarn test:changed --browser=firefox
+
+            - name: Create a dummy file to ensure at least one results file exists
+              run: touch tachometer.firefox-ran.txt
+
+            - name: Archive Firefox tachometer results
+              uses: actions/upload-artifact@v4
+              with:
+                  name: tachometer-results-firefox
+                  path: |
+                      tach-results.firefox.*.json
+                      tachometer.firefox-ran.txt
+
+    compare-chrome:
```

Comment:
Looking at this, it seems the jobs are exactly the same.
Could we try defining a [matrix](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow) for this?
It could probably reduce this file by half and the tests run in parallel 😃

Reply:
Can that be a separate ticket? This was scoped to fixing smoke

Reply:
I am fine with this - this is not required